editor/editor_node.cpp

```diff
@@ -7420,6 +7420,19 @@ EditorNode::EditorNode() {
 		file_menu->add_shortcut(ED_SHORTCUT_AND_COMMAND("editor/file_quit", TTR("Quit"), KeyModifierMask::CMD_OR_CTRL + Key::Q), FILE_QUIT, true);
 	}
 
+	ED_SHORTCUT_AND_COMMAND("editor/editor_settings", TTR("Editor Settings..."));
+	ED_SHORTCUT_OVERRIDE("editor/editor_settings", "macos", KeyModifierMask::META + Key::COMMA);
+#ifdef MACOS_ENABLED
+	if (global_menu) {
+		apple_menu = memnew(PopupMenu);
+		apple_menu->set_special_menu_root("_apple");
+		apple_menu->add_shortcut(ED_GET_SHORTCUT("editor/editor_settings"), SETTINGS_PREFERENCES);
+		apple_menu->add_separator();
```

Comment:
There are no more items that follow, why do we need a separator? Or are items added by the display server automatically?

Reply:
There are a lot of standard items added by OS (both before and after this).

Reply:
So in this case we use popup menus as a interface and data container to populate some pre-existent system menus?

Reply:
Yes. We are adding this part:
<img width="272" alt="Screenshot 2023-12-13 at 15 32 02" src="https://github.com/godotengine/godot/assets/7645683/017da579-15de-45b0-b2e0-2f86f0017e2a">